app/controllers/SupplementaryUnitController.scala

```diff
@@ -53,15 +54,31 @@ class SupplementaryUnitController @Inject() (
         case Some(value) => form.fill(value)
       }
 
-      Ok(view(preparedForm, mode, recordId))
+      request.userAnswers.get(RecordCategorisationsQuery) match {
+        case None        => Redirect(routes.JourneyRecoveryController.onPageLoad().url)
+        case Some(query) =>
+          query.records(recordId).measurementUnit match {
```

Comment:
not sure we can/should be assuming
query.records(recordId)
is defined?

Could be worth switching to a for/yield like in the other places we get the details out of the recordCategorisationsQuery

Reply:
Updated to for/yield